### PR TITLE
implement `/eth/v1/beacon/blocks` post rest api

### PR DESCRIPTION
**Pull request by rolfyone**
- deprecated `POST /validator/block`

 - the functionality is identical, just on a different endpoint, so moved the test suite to the new location and test the new class.

 partially addresses #2756

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  private String handleResponseContext(
+      final Context ctx, final ValidatorBlockResult validatorBlockResult)
+      throws JsonProcessingException {
+    ctx.status(validatorBlockResult.getResponseCode());
+    if (validatorBlockResult.getFailureReason().isPresent()) {
+      return jsonProvider.objectToJSON(validatorBlockResult.getFailureReason().get());
+    } else {
+      return jsonProvider.objectToJSON(
+          validatorBlockResult.getHash_tree_root().map(Bytes32::toHexString).orElse(""));
+    }
```
Comment (ajsutton) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java]: I don't think this matches the expected output.  It looks like the new API has no content for either 200 or 202 and error responses should have a code and message.

Comment (ajsutton) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java]: Can we also add some tests to cover what this should output please? I discovered on the weekend we had none and it was actually converting an `Optional` to JSON which wasn't pretty.

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java]: Hrm I just lifted and shifted, i'll take another pass.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java [hunk 2]
```diff
@@ -34,12 +35,13 @@
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
-import org.apache.tuweni.bytes.Bytes32;
+import joptsimple.internal.Strings;
```
Comment (ajsutton) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java]: er, this looks like an odd import.  I think I'd just use `""` for an empty string, seems pretty easily understandable.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java (hunk 2 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java]: wow that is weird.. i thought strings.Empty was pretty standard, i'll fix. ta
